libsentrykube/quickpatch.py

```diff
@@ -1,15 +1,122 @@
-from typing import Sequence, Mapping
+import os
+from typing import MutableMapping, Optional, Sequence, Union
+import yaml
+import jsonpatch
+from jinja2 import Template
+
+from libsentrykube.service import (
+    get_service_path,
+    get_service_value_overrides_file_path,
+)
+from jsonschema import validate, ValidationError
+
+
+def find_patch_files(service: str, patch: str) -> Optional[str]:
+    """
+    Finds the patch file for the given service and patch name
+    """
+    base_path = get_service_path(service)
+    print(f"Base path: {base_path}")
+    for root, dirs, files in os.walk(base_path):
+        print(f"Root: {root}, dirs: {dirs}, files: {files}")
+        if "quickpatches" in root:
+            for file in files:
+                if file.endswith(f"{patch}.yaml.j2"):
+                    return os.path.join(root, file)
```

Comment:
THis is simpler:

```
from pathlib import Path

expected_path = Path(base_path) / "quickpatches"

if  expected_path.exists() and expected_path.is_dir():
    patch_file = expected_path / f"{patch}.j2"
    if patch_file.exists(): 
       return patch_file
return None 
```

See a lot of usages of pathlib.Path https://github.com/getsentry/sentry-infra-tools/blob/main/config_builder/combined_generator.py